### PR TITLE
Migrate to ODS & Enable Read,Write,Header Commands

### PR DESCRIPTION
**Pull request by chocorho**
This update will start a major rewrite of the core of `gspreadtest`, the main
building-block of the spreadsheet operations.

It will also start refining
the code for the databot itself, which depends on gspreadtest.

The new variant of `gspreadtest` is designed to follow the original
specification as closely as possible (using the same method names whenever
possible), but it will instead use **ODS** files and **Nextcloud** as opposed to
Google Drive documents.

More testing & development work is needed, but the reader will notice that the basic tests
depicted in the new `spreadsh_mgr` file should be functional.

Let us be free and control our own data!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -87,17 +107,17 @@ async def newHeader(message, *args):
           await message.channel.send(f'You can\'t make two headers of the same name. \n(if ur trying to find bugs id like you to know ur an asshole)')
           
         else:
-          spreadSheet.write_cell(f'{string.ascii_uppercase[count]}1', headers)
+          self.spreadsheet.write_cell(f'{string.ascii_uppercase[count]}1', headers)
```
Comment (chocorho) [inline main.py]: WARNING: this implementation is successful if and only if `count` is within the bounds of `ascii_uppercase`, i.e. between 0 and 25. We should either (a) remove this limitation and enable spreadsheets with more than 26 headers (I can rewrite the formula to allow this) or (b) do index-checking before running this else-case, and print an error to the user in an effort to fail gracefully.
@megumin00 which do you prefer?

File: main.py [hunk 2]
```diff
@@ -51,30 +74,27 @@ async def ping(message):
...
         
         #compiling arguments in message contents
-        for i in args:
-          if i != args[-1]:
-            headers += str(i) + ' '
-        headers += args[-1]
+        headers = " ".join(str(next_header) for next_header in args)
         
-        #find every second column so that I can fit datetine values for each cell (B, D, F, etc are all data columns while the other odds contain dates)
+        #find every second column so that I can fit datetime values for each cell (B, D, F, etc are all data columns while the other odds contain dates)
```
Comment (chocorho) [inline main.py]: Can you clarify this comment further? How do we know that each cell in columns B, D, F, H, ... will need its own corresponding timestamp? Won't it depend on what data is stored in the given cell, and therefore won't it depend on the header set by the user?
@megumin00 I'd appreciate a clarification of the requirements. If the bot is logging user messages, then I imagine the messages would be saved in column B with their timestamps in column A (right?). But what will be stored in column D that _also_ needs a timestamp separate from the timestamp given in column A at the same row number? I'd like to see an example.

**Comment by chocorho** [comment]
Well, I could resolve the Merge Conflicts myself, but I want to give you some practice with it. :stuck_out_tongue: 

Oh, and I want you to review the PR, too (~~ultierior motive~~).

@megumin00

**Review by suwuako — COMMENTED** [review]
fair enough

**Pull request merged**